src/inference/src/dev/core_impl.hpp

```diff
@@ -90,33 +90,60 @@ class CoreImpl : public InferenceEngine::ICore, public std::enable_shared_from_t
+        bool core_config_supported(const ov::Plugin& plugin, const std::string& config_name) const;
+
+        // Clean up core config for input config.
+        // If 'update' is true, this core config will be update into config cache.
+        void remove_core_config(const ov::Plugin& plugin, ov::AnyMap& config, const bool update = false) const;
+
+        // Update input config with the help of core_plugins_properties:
+        //  1. Remove the core config that this plugin doesn't support
+        //  2. Don't overwritten this config if they have been in input config
+        //  3. Add default core config into input config if it doesn't have.
+        template <typename T>
+        void update_input_config(ov::Plugin& plugin, std::map<std::string, T>& config) const;
+
+        std::set<std::string> query_core_config_keys() const;
 
         // Creating thread-safe copy of config including shared_ptr to ICacheManager
         // Passing empty or not-existing name will return global cache config
-        CacheConfig get_cache_config_for_device(const std::string& device_name,
-                                                bool device_supports_cache_dir,
-                                                ov::AnyMap& parsedConfig) const;
+        CoreConfigCache get_cache_config_for_device(const ov::Plugin& plugin,
+                                                    const std::string& config_name,
+                                                    ov::AnyMap& parsedConfig) const;
 
-        CacheConfig get_cache_config_for_device(const std::string& device_name) const;
+        // Debug code, will remove later.
+        void print_core_properties() const;
 
     private:
-        static void fill_config(CacheConfig& config, const std::string& dir);
+        static void fill_config(CoreConfigCache& config, const std::string& key, const std::string& value);
 
-        mutable std::mutex _cacheConfigMutex;
-        CacheConfig _cacheConfig;
-        std::map<std::string, CacheConfig> _cacheConfigPerDevice;
+        mutable std::mutex _CoreConfigCacheMutex;
+        mutable std::map<std::string, CoreConfigCache> _core_properties_cache_per_device;
+
+        // Core global properties, which will not be set to any plugins.
+        // It will be updated if core.set_property() without device name is called.
+        mutable CoreConfigCache _core_global_properties = {{{ov::force_tbb_terminate.name(), ov::Any(false)}}, nullptr};
+
+        // Core plugins properties, which will be set to all plugins.
+        // It will be updated if core.set_property() is called.
+        mutable CoreConfigCache _core_plugin_properties = {{{ov::cache_dir.name(), ""},
+                                                            {ov::hint::allow_auto_batching.name(), ov::Any(true)},
+                                                            {ov::auto_batch_timeout.name(), "1000"}},
+                                                           nullptr};
```

Comment:
I would like to avoid more and more types of properties and handlers for these properties in ov::Core. As I already said, looks like it can be solved in different manner.

Reply:
How do you think to create `ov::core_properties` property object like below to manage all core properties?
`     static constexpr Property<std::map<std::string, std::string>, PropertyMutability::RW> core_properties{"CORE_PROPERTIES"};`

User and plugin both can use this `ov::core_properties` to get and set it.